cmake/modules/FindMaya.cmake

```diff
@@ -407,6 +407,15 @@ if (MAYA_API_VERSION VERSION_GREATER_EQUAL 20230000)
     set(MAYA_ENABLE_NEW_PRIM_DELETE TRUE CACHE INTERNAL "enableNewPrimDelete")
 endif()
 
+set(MAYA_ARRAY_ITERATOR_DIFFERENCE_TYPE_SUPPORT FALSE CACHE INTERNAL "hasArrayIteratorDifferenceType")
+if(MAYA_INCLUDE_DIRS AND EXISTS "${MAYA_INCLUDE_DIR}/maya/MArrayIteratorTemplate.h")
+    file(STRINGS ${MAYA_INCLUDE_DIR}/maya/MArrayIteratorTemplate.h MAYA_HAS_API REGEX "hasArrayIteratorDifferenceType")
```

Comment:
The regular expression here is not correct. This line is going to search MArrayIteratorTemplate.h for the string "hasArrayIteratorDifferenceType" and set MAYA_HAS_API if it does. However, "hasArrayIteratorDifferenceType" is not in MArrayIteratorTemplate.h and so we'll never set MAYA_HAS_API true.

I think the REGEX to search for should be "difference_type". That is a string which is in the updated file and not in the older file.

Reply:
You are correct, I accidentally uploaded this bugged and incomplete (I will add the new env variable at the top of the file comment) version of the code review change.